### ISSUE TITLE
Exception branches are still counted after the parameter "--exclude-throw-branches" is used

### ISSUE THREAD

**Issue by SimonsSoft** [gcovr/gcovr#957]
# Describe the bug

Exception branches are still counted after the parameter "--exclude-throw-branches" is used

# To Reproduce

1. target test code is this:
```Cpp
static ara::core::Result<ProcessSpecifier> Create(ara::core::StringView name)
    {
        ara::core::Result<void> ret = Validate(name);
        if ( !ret )
        {
            return ara::core::Result<ProcessSpecifier>::FromError(ara::exec::FindProcessErrc::kInvalidArgument);
        }

        ProcessSpecifier result{};
        result.name_ = name;
        return result;
    }
```

2. with gcovr command "gcovr --html coverage/index.html --html-details -r ${srcdir}  --exclude-throw-branches --exclude-unreachable-branches --object-directory=${objdir} --exclude-function-lines --exclude-noncode-lines -k --verbose "

# Expected behavior

Expected 100% coverage. But throw branch have counted.

# Screenshots

1. html report:
![image](https://github.com/user-attachments/assets/a2e87ff4-b0f9-4d76-8e29-4547f12e32ae)

2. gcov file info:
![image](https://github.com/user-attachments/assets/28adb7b4-0d1d-48c5-9a64-d0e9d179a51b)

3. some gcovr report not match gcov file:
![image](https://github.com/user-attachments/assets/a7e7f63f-85cb-46ee-859f-7f67bf1450fc)

5. verbose log:
[gcovr.log](https://github.com/user-attachments/files/16886264/gcovr.log)

# Desktop (please complete the following information)

- OS: Ubuntu 20.04.6 LTS
- GCC version: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
- GCOVR version: 7.2
- Project directory layout:
  - Roots
  - Object directories

# Additional context

I want to reach 100% coverage no matter what method I use, Can you help me, Thank you very much!

**Comment by Spacetown** [comment]
Thanks for reporting this. Is it possible to check the infos written to the JSON output format?

**Comment by SimonsSoft** [comment]
> Thanks for reporting this. Is it possible to check the infos written to the JSON output format?
I am very glad to receive your reply. this is the json output file.
[gcovr_coverage.json](https://github.com/user-attachments/files/16886619/gcovr_coverage.json)

**Comment by Spacetown** [comment]
In this JSON are both branches (line 62 and 65) are not marked as `throw` but the count is different from your gcov test output. Was the JSON generated with the same version of gcc?
There was a bug in gcov where the `(throw)` was missing, see https://github.com/gcovr/gcovr/issues/677#issuecomment-1284417325.

**Comment by SimonsSoft** [comment]
Thank you very much!
After I installed gcc 13.1 with reference to #677, retest OK.

![image](https://github.com/user-attachments/assets/44ca60ef-9da4-42c9-a155-92913c9adf3b)

**Comment by kfwalther** [comment]
I'm experiencing this same issue on Debian 11 with GCC 10.2. We have a ton of degraded branch coverage metrics due to `throw` branches not being tested. We'd like to ignore these cases, but adding `--exclude-throw-branches` has no effect. Is the only fix to upgrade all the way to GCC 13.1+ ?

**Comment by Spacetown** [comment]
@kfwalther if the lines have a common pattern you can write a regular expression to ignore all the lines by pattern.
